server/jetstream_test.go

```diff
@@ -24365,3 +24365,25 @@ func TestJetStreamStreamConfigClone(t *testing.T) {
 	clone.Metadata["key"] = "value"
 	require_False(t, reflect.DeepEqual(cfg.Metadata, clone.Metadata))
 }
+
+func TestIsJSON(t *testing.T) {
+	tests := []struct {
+		name  string
+		data  []byte
+		valid bool
+	}{
+		{"empty", []byte{}, false},
+		{"empty_object", []byte("{}"), true},
+		{"empty object, not trimmed", []byte("\t\n\r{ }"), true},
+		{"empty_array", []byte("[]"), true},
+		{"empty array, not trimmed", []byte("\t\n\r[ ]"), true},
+		{"empty_string", []byte("\"\""), false},
```

Comment:
Perhaps this is out of scope, but "" is a valid JSON marshal of just an empty string.

Reply:
This function accepts only JSON objects and JSON arrays, hence the test. Added more tests, renamed function, added better docs. Let me know if it makes sense now.